x-pack/legacy/plugins/security/server/lib/authentication/authenticator.ts

```diff
@@ -56,11 +57,14 @@ function assertRequest(request: Legacy.Request) {
  */
 function getProviderOptions(server: Legacy.Server) {
   const config = server.config();
+  const client = getClient(server);
+  const log = server.log.bind(server);
 
   return {
-    client: getClient(server),
-    log: server.log.bind(server),
+    client,
+    log,
     basePath: config.get<string>('server.basePath'),
+    tokens: new Tokens({ client, log }),
```

Comment:
__note:__ `basic` is the only provider that doesn't use tokens and it's pretty cheap to instantiate it even if `basic` provider is used, so I decided to have it as provider option instead of re-creating it for every provider that uses tokens.